lib/sles4sap.pm

```diff
@@ -284,9 +284,32 @@ sub test_start_instance {
     my $output = script_output "sapcontrol -nr $instance -function Start";
     die "sapcontrol: Start API failed\n\n$output" unless ($output =~ /Start[\r\n]+OK/);
 
-    $output = script_output $ps_cmd;
-    my @olines = split(/\n/, $output);
-    die "sapcontrol: failed to start the instance" unless (@olines > 1);
+    my $time_to_wait = get_var('WAIT_INSTANCE_START', 300);    # Wait by default for 5 minutes
+    $time_to_wait = 600 if ($time_to_wait > 600);              # Limit this to 10 minutes max
+
+    while ($time_to_wait > 0) {
+        $output = script_output "sapcontrol -nr $instance -function GetSystemInstanceList";
+        die "sapcontrol: GetSystemInstanceList: command failed" unless ($output =~ /GetSystemInstanceList[\r\n]+OK/);
+
+        if ($output =~ /GREEN/) {
+            $output = script_output "sapcontrol -nr $instance -function GetProcessList | egrep -i ^[a-z]", proceed_on_failure => 1;
+            die "sapcontrol: GetProcessList: command failed" unless ($output =~ /GetProcessList[\r\n]+OK/);
+
+            my $failing_services = 0;
+            for my $line (split(/\n/, $output)) {
+                next if ($line =~ /GetProcessList|OK|^name/);
+                $failing_services++;
+                $failing_services-- if ($line =~ /GREEN/);
```

Comment:
```suggestion
                $failing_services++ if ($line !~ /GREEN/);
```

Reply:
You're correct. Will commit and test.

Reply:
New verification run: http://mango.suse.de/tests/1293

Reply:
Pushed changes. @ricardobranco777 please review again